File: scripts/sync-from-main.mjs
import { execSync } from 'child_process';

function run(cmd) {
  console.log(`> ${cmd}`);
  const output = execSync(cmd, { cwd: '/vercel/share/v0-project', encoding: 'utf8' });
  if (output) console.log(output);
  return output;
}

try {
  run('git fetch origin main');
  run('git merge origin/main --no-edit');
  console.log('Successfully synced latest from origin/main.');
} catch (err) {
  console.error('Error syncing from main:', err.message);
  process.exit(1);
}

File: scripts/sync-from-main.py
import subprocess
import sys

def run(cmd):
    print(f"Running: {cmd}")
    result = subprocess.run(cmd, shell=True, capture_output=True, text=True, cwd="/vercel/share/v0-project")
    if result.stdout:
        print(result.stdout)
    if result.stderr:
        print(result.stderr)
    if result.returncode != 0:
        print(f"Command failed with exit code {result.returncode}")
        sys.exit(result.returncode)
    return result.stdout

run("git fetch origin main")
run("git merge origin/main --no-edit")
print("Successfully synced from origin/main.")
